src/v/storage/spill_key_index.cc

```diff
@@ -87,6 +87,8 @@ ss::future<> spill_key_index::add_key(bytes b, value_type v) {
     auto const key_size = b.size();
     auto const expected_size = idx_mem_usage() + _keys_mem_usage + key_size;
 
+    // TODO call into storage_resources
```

Comment:
something needed in this PR?

Reply:
I'd like to do that in a followup before 22.2.1, https://github.com/redpanda-data/redpanda/issues/5389

The reason I didn't already do it is because the testing isn't covering compaction well yet, so if I make a change now it'll be guesswork as to whether it's sufficient to get robustness.